lib/tqueue.ml

```diff
@@ -0,0 +1,65 @@
+type 'a t = {
+  queue : 'a Queue.t;
+  lock : Lwt_mutex.t;
+  has_elt : unit Lwt_condition.t;
+}
+
+let create () =
+  let queue = Queue.create () in
+  let lock = Lwt_mutex.create () in
+  let has_elt = Lwt_condition.create () in
+  { queue; lock; has_elt }
+
+let add x { queue; lock; has_elt } =
+  Lwt_mutex.with_lock lock (fun () ->
+      Queue.add x queue;
+      let%lwt () =
+        if Queue.length queue = 1 then (
+          Lwt_condition.signal has_elt ();
+          Lwt.return ())
+        else
+          Lwt.return () in
+      Lwt.return ())
+
+let push = add
```

Comment:
What's the purpose of these? Why simply rename `add` to `push`? 🤔

Reply:
Just copying https://ocaml.org/api/Queue.html for minimal friction :smile:

Reply:
So maybe we can get rid of them, don't you think?

Reply:
Sure :slightly_smiling_face: I wasn't planning on using them

Reply:
They correspond to the safe versions of the functions now, so I don't see any issue with them.